mitxgraders/formulagrader/matrixgrader.py

```diff
@@ -82,11 +98,29 @@ def schema_config(self):
             }): {
                 Required('is_raised', default=True): bool,
                 Required('msg_detail', default='type'): Any(None, 'type', 'shape')
-            }
+            },
+            Required('entry_partial_credit', default=0): Any(All(Number, Range(0, 1)), 'proportional'),
+            Required('entry_partial_msg', default=MatrixEntryComparer.default_msg): text_string
         })
 
+    @classmethod
+    def get_default_comparer(cls, unvalidated_config):
+        keys = ['entry_partial_credit', 'entry_partial_msg']
+        comparer_config = {key: unvalidated_config[key]
+                           for key in keys if key in unvalidated_config}
+        if comparer_config:
+            return MatrixEntryComparer(comparer_config)
+        return cls.default_comparer
+
     def __init__(self, config=None, **kwargs):
+        # Set default_comparer on as an instance property
+        # This will be used by schema_config during validation to set defaults
+        unvalidated_config = config if config is not None else kwargs
```

Comment:
Is there a reason not to do this after the `super` call? That way, the config is validated...

Reply:
The reason to do this before `super` is that the `super` call will use `self.default_comparer` during validation to provide a comparer function as part of `answers` (if a comparer function was not explicitly provided). 

This portion of the config is still validated—twice, actually. It's validated when `super(MatrixGrader, self).__init__` runs, and it is also validated when the comparer function instantiates .